daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -367,11 +396,13 @@ convertChoice env (VarIs "C:Choice" `App` Type tmpl `App` Type (TypeCon chc [])
 
         f i (App a _) = f i a
         f i (Tick _ e) = f i e
-        f i (VarIs "nonconsuming") = pure False
-        f i (VarIs "$dmconsuming") = pure True -- the default is consuming
+        f i (VarIs "$dmconsuming") = pure (True, False) -- the default is preconsuming
```

Comment:
Can we have a proper Enum introduced and return Preconsuming/Postconsuming/Nonconsuming here and then `==` for the tests above.

Reply:
✔️